### DISCUSSION TITLE
Hidden cross-package (workspace) dependencies

### DISCUSSION THREAD

**Discussion by marek-hanzal** [vercel/turborepo#2259] [Help] [answered]
Hello,

I'm doing something wrong, but the question is: I've several `/packages/[here 1, here 2, and here 3]`. The strange thing I'm experiencing is when I put a dependency in `here 2/package.json`, it's also available in all the other packages. With this I'm cannot be sure a package is standalone as it does not have all it's dependencies.

How I could solve this? I need to have separated `package.json` for each package, not merged and shared them together.

Thank you!

Example of one package without specified dependency in `package.json`
![image](https://user-images.githubusercontent.com/13855126/196428881-ffade91e-0460-462d-b3c6-b64f704e9ce3.png)
...is happily using it
![image](https://user-images.githubusercontent.com/13855126/196429014-a276e835-fd8e-4803-bd12-8945552cdffe.png)

**Comment by mehulkar** [comment, marked as answer]
This is not related to turborepo. My guess is that when you install a dependency in `packages/here 2`, the package is installed in the root `node_modules/` directory. _Most likely_ your other packages have the ability to import/require packages from the root. I am not sure what patterns there are to solve this, however. It's possible that the [`npm` docs on workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces#adding-dependencies-to-a-workspace) (or the docs of whatever package manager you're using) covers this!
